pyinfra/operations/selinux.py

```diff
@@ -0,0 +1,177 @@
+        host.noop(f"seboolean '{boolean}' already had the value '{value}'")
+
+
+@operation(
+    pipeline_facts={"filecontext": "path"},
+)
+def file_context(path, se_type):
+    """
+    Set the SELinux type for the specified path to the specified value.
+
+    + path: the target path (expression) for the context
+    + se_type: the SELinux type for the given target
+
+    **Example:**
+
+    .. code:: python
+
+        selinux.file_context(
+            name = 'Allow /foo/bar to be served by the web server',
+            '/foo/bar',
+            'httpd_sys_content_t'
+        )
+    """
+
+    current = host.get_fact(FileContext, path=path) or {}
+    if se_type != current.get("type", ""):
+        yield StringCommand("chcon", "-t", se_type, QuoteString(path))
+        host.create_fact(
+            FileContext,
+            kwargs={"path": path},
+            data=dict(current, **{"type": se_type}),
+        )
+    else:
+        host.noop(f"file_context: '{path}' already had type '{se_type}'")
+
+
+@operation(
+    pipeline_facts={"filecontextmapping": "target"},
+)
+def file_context_mapping(target, se_type=None, present=True):
+    """
+    Set the SELinux file context mapping for paths matching the target.
+
+    + target: the target path (expression) for the context
+    + se_type: the SELinux type for the given target
+    + present: whether to add or remove the target -> context mapping
+
+    Note: `file_context` does not change the SELinux file context for existing files
+    so `restorecon` may need to be run manually if the file contexts cannot be created
+    before the related files.
+
+    **Example:**
+
+    .. code:: python
+
+        selinux.file_context_mapping(
+            name = 'Allow Apache to serve content from the /web directory',
+            r'/web(/.*)?',
+            se_type='httpd_sys_content_t'
+        )
+    """
+    if present and (se_type is None):
+        raise ValueError("se_type must have a valid value if present is set")
+
+    current = host.get_fact(FileContextMapping, target=target)
+    kwargs = {"target": target}
+    if present:
+        op = "-a" if len(current) == 0 else ("-m" if current.get("type") != se_type else "")
+        if op != "":
+            yield StringCommand("semanage", "fcontext", op, "-t", se_type, QuoteString(target))
+            host.create_fact(
+                FileContextMapping,
+                kwargs=kwargs,
+                data=dict(current, **{"type": se_type}),
+            )
+        else:
+            host.noop(f"mapping for '{target}' -> '{se_type}' already present")
+    else:
+        if len(current) > 0:
+            yield StringCommand("semanage", "fcontext", "-d", QuoteString(target))
+            host.create_fact(FileContextMapping, kwargs=kwargs, data={})
+        else:
+            host.noop(f"no existing mapping for '{target}'")
+
+
+@operation(
+    pipeline_facts={"seport": "protocol, port"},
+)
+def se_port(protocol, port, se_type=None, present=True):
```

Comment:
```suggestion
def port(protocol, port, se_type=None, present=True):
```

Same as above for boolean, `selinux.port`...

Reply:
changed